### PR TITLE
feat(Light Novel Pub): add presence

### LINKED ISSUE #5916: lightnovelpub | www.lightnovelpub.com/web
### Website name

LIGHT NOVEL PUB

### Website URL

https://www.lightnovelpub.com/web

### Website logo

https://static.lightnovelpub.com/content/img/lightnovelpub/logo.png

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

Reading "novel name", chapter you are on and max count "ch142/2000" and button that you can click that sends you to the novel on the site.

reading:
Novel name,
chapter 123/1230,
button to novel on site.

### PR DESCRIPTION
**Pull request by ghost**
**Describe the changes in this pull request:** Resolves #5916
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

<br>

<br>

</details>

### CODE REVIEW

**Comment by EncryptedDev** [comment]
Hi there! What is the reason for this being a draft?

**Comment by yrnmsk** [comment]
> Hi there! What is the reason for this being a draft?

i havent completely finished this presence yet as i was having troubles locally with my premid and the premid docs so i had no choice but to commit the changes and go to bed

**Comment by EncryptedDev** [comment]
> > Hi there! What is the reason for this being a draft?
> 
> i havent completely finished this presence yet as i was having troubles locally with my premid and the premid docs so i had no choice but to commit the changes and go to bed
> 
> `edit:` turns out, premid docs is actually currently down, but i was already having troubles with premid hours before this, so not really sure

Feel free to reach out to me on discord for help (:

**Comment by yrnmsk** [comment]
> Feel free to reach out to me on discord for help (:

thanks, will do! but i still have univ stuffs for my whole day ahead, so maybe after univ stuffs, premid and docs works fine again anyways

**Comment by yrnmsk** [comment]
i changed this from draft as this is ready. i also edited [this message](https://github.com/PreMiD/Presences/pull/5941#issue-1168602215) as the images there were also drafts

File: websites/L/Light Novel Pub/presence.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			details: "Browsing...",
+			startTimestamp: browsingStamp
+		};
+
+	if (pathname.startsWith("/novel/")) {
+		presenceData.largeImageKey = document
+			.querySelector('meta[property="og:image"]')
+			.getAttribute("content");
+		presenceData.smallImageKey = "logo";
+		presenceData.startTimestamp = browsingStamp;
```
Comment (Slowlife01) [inline websites/L/Light Novel Pub/presence.ts:18]:
```suggestion
```

File: websites/L/Light Novel Pub/presence.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		presenceData.buttons = [
+			{
+				label: "",
```
Comment (Slowlife01) [inline websites/L/Light Novel Pub/presence.ts:21]: ?

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

File: websites/L/Light Novel Pub/presence.ts (hunk 2 above)
Comment (yrnmsk) [reply websites/L/Light Novel Pub/presence.ts:21]: i edited it again down below cuz i usually set the button's label just the same with either `.Details` or `.State` so... should i not?

**Review by EncryptedDev — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**